src/site.rs

```diff
@@ -33,24 +35,41 @@ impl Site {
         self.options.is_valid()
     }
 
-    pub fn to_file_base_name(&self) -> String {
-        self.options
-            .filename
-            .clone()
-            .unwrap_or_else(|| filename::basename(&self.url))
+    pub fn to_file_path(&self) -> PathBuf {
+        self.options.filename.clone().unwrap_or_else(|| {
+            let folder = filename::domainfolder(&self.url);
+            let [first, rest @ ..] = &folder[..] else {
+                panic!(
+                    "domain has to have at least one segment {folder:?} {:?}",
```

Comment:
The surrounding code challenges my skill level of Rust. It might be wise to add a comment to keep the barrier of entry low? Additionally, do we have a unittest that checks for having at least one segment (or the reverse)? What does that even mean exactly?

Reply:
The domain is either something like `edjopato.de` which has at least two segments (`de` and `edjopato`) or is an IP which is one segment. As you have a target to reach you always need at least some information on how to reach that target so there is something in that domain / host.

Maybe the confusing part is the `panic` here and it should be a `unreachable`.

There are unit tests in the filename module.

The let else is basically a different way of if let else but with just exposing the result to the following code. The code could also be written as a match:

```rust
match &folder[..] {
  [first, rest @ ..] => todo!("use"),
  [] => unreachable!(),
}
```

The slice has basically two ways to be: has at least one element (`first`) (`rest` can be 0 or longer) or is empty. Empty is not possible here (or a bug, not something that should happen on runtime).